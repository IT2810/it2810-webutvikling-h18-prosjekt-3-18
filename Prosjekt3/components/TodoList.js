import React, { Component } from 'react';
import { View, Text, Button, TouchableOpacity} from 'react-native';
import ProgressBar from "./ProgressBar";

class TodoList extends Component {
    constructor(props) {
        super(props);
        this.onPressOpen = this.onPressOpen.bind(this);

    }

    render() {
        return (
            <View style={{ flexDirection: 'row', flex:0.95, marginLeft:2, paddingTop:2, marginTop:2,  alignItems: 'stretch'}}>
                <TouchableOpacity title="Open" onPress={this.onPressOpen}>
                <Text style={{width:100}}> {this.props.name}</Text>
                </TouchableOpacity>
                <TouchableOpacity title="Open" onPress={this.onPressOpen}>
                    <View style={{alignItems: 'center', marginLeft: 40, paddingTop: 5}}>
                    <ProgressBar
                        current={this.props.complete}
                        max={this.props.total}
                        height={10}
                        width={100}/>
                    </View>
                </TouchableOpacity>
            </View>

        )
    }

    onPressOpen() {
        this.props.onOpen(this.props.id);
    }
}

export default TodoList;
