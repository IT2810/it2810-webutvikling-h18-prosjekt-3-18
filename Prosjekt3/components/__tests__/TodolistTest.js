import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import TodoList from '../../components/TodoList';

it('renders correctly', () => {
    const tree = renderer.create(<TodoList>Snapshot test!</TodoList>).toJSON();
    expect(tree).toMatchSnapshot();
});
