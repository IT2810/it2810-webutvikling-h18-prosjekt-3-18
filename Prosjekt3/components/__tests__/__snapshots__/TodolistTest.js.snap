// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "stretch",
      "flex": 0.95,
      "flexDirection": "row",
      "marginLeft": 2,
      "marginTop": 2,
      "paddingTop": 2,
    }
  }
>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "width": 100,
        }
      }
    >
       
    </Text>
  </View>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "marginLeft": 40,
          "paddingTop": 5,
        }
      }
    >
      <View>
        <View
          animated={true}
          animationConfig={
            Object {
              "bounciness": 0,
            }
          }
          animationType="spring"
          border-radius="12px"
          indeterminate={false}
          onLayout={[Function]}
          progress={NaN}
          style={
            Array [
              Object {
                "backgroundColor": "#F22",
                "borderColor": "#000",
                "borderRadius": 4,
                "borderWidth": 1,
                "overflow": "hidden",
                "width": 100,
              },
              undefined,
            ]
          }
          useNativeDriver={false}
        >
          <View
            style={
              Object {
                "backgroundColor": "#0F0",
                "height": 10,
                "transform": Array [
                  Object {
                    "translateX": 0,
                  },
                  Object {
                    "translateX": NaN,
                  },
                  Object {
                    "scaleX": NaN,
                  },
                ],
              }
            }
          />
        </View>
      </View>
    </View>
  </View>
</View>
`;
